compiler/src/dotty/tools/dotc/typer/Namer.scala

```diff
@@ -712,7 +712,7 @@ class Namer { typer: Typer =>
               enterSymbol(classConstructorCompanion(classSym.asClass))
             else
               for moduleSym <- companionVals do
-                if moduleSym.is(Module) && !moduleSym.isDefinedInCurrentRun then
+                if moduleSym.lastKnownDenotation.is(Module) && !moduleSym.isDefinedInCurrentRun then
```

Comment:
Can you add a little detail in a comment about this usage (with any references that would be helpful)